### PR TITLE
docs: tutorial on using Lets Encrypt for automatic certificate issuance and rotation

### PR DESCRIPTION
**Pull request by khowling**
Updated the docs with a new section in the tutorial to describe setting up 'Certificate issuance with Lets Encrypt'

### CODE REVIEW

**Comment by asridharan** [comment]
This is really cool @khowling  💯 !!

@draychev  @akshaysngupta could you PTAL !!

File: docs/tutorial.md [hunk 1]
```diff
@@ -160,6 +161,107 @@ By specifying hostname, the guestbook service will only be available on the spec
...
 Now the `guestbook` application will be available on both HTTP and HTTPS only on the specified host (`<guestbook.contoso.com>` in this example).
 
+### Certificate issuance with Lets Encrypt
+
+This section configures [Lets Encrypt](https://letsencrypt.org/) to issue the certificate that Applicaiton Gateway will use for SSL/TLS termination at the gateway.  This uses the Kubernetes add-on [cert-manager](https://github.com/jetstack/cert-manager) to automate the creation and management of the kubernetes secret containing the certificate.
+
+> [!NOTE]
+> The following steps will install `cert-manager` in your cluster, full documnetation can be found on the cert-manager docs site [here](https://docs.cert-manager.io)
+
+1. run this script to install the `cert-manager` helm chart [(from docs.cert-manager.io)](https://docs.cert-manager.io/en/latest/getting-started/install/kubernetes.html#steps)
```
Comment (akshaysngupta) [inline docs/tutorial.md]: nit: Execute

File: docs/tutorial.md [hunk 2]
```diff
@@ -160,6 +161,107 @@ By specifying hostname, the guestbook service will only be available on the spec
...
+
+# Install the cert-manager Helm chart
+helm install \
+  --name cert-manager \
+  --namespace cert-manager \
+  --version v0.8.0 \
+  jetstack/cert-manager
+```
+
+2. Create a `ClusterIssuer`, these details are required by `cert-manager` to automatically generate a certificate request which is fulfilled by the issuer (`Lets Encrypt`).  `Let’s Encrypt` uses the ACME protocol to verify that you control a given domain name and to issue you a certificate
```
Comment (akshaysngupta) [inline docs/tutorial.md]: Where is the `ClusterIssuer` resource suppose to reside ?

File: docs/tutorial.md [hunk 3]
```diff
@@ -160,6 +161,107 @@ By specifying hostname, the guestbook service will only be available on the spec
...
+
+> [!NOTE]
```
Comment (akshaysngupta) [inline docs/tutorial.md]: `[!NOTE]` is this supported ?

File: docs/tutorial.md [hunk 4]
```diff
@@ -160,6 +161,107 @@ By specifying hostname, the guestbook service will only be available on the spec
...
+      paths:
+      - backend:
+          serviceName: frontend
+          servicePort: 80
+```
+
+
+After a few seconds, you  can access the `guestbook` service through the Application Gateway HTTPS url using the automatically issued `Lets Encrypt` certificate. 
+
+NOTE: Before the `Lets Encrypt` certificate expires, `cert-manager` will  update the certificate in the kubernetes secret store. At that point, application-gateway-ingress-controller will apply the updated secret referenced in the ingress resources it is using to configure the Application Gateway.
```
Comment (akshaysngupta) [inline docs/tutorial.md]:
```
> **_NOTE:_**  The note content.
```
> **_NOTE:_**  The note content.

File: docs/tutorial.md [hunk 5]
```diff
@@ -160,6 +161,107 @@ By specifying hostname, the guestbook service will only be available on the spec
 
 Now the `guestbook` application will be available on both HTTP and HTTPS only on the specified host (`<guestbook.contoso.com>` in this example).
 
+### Certificate issuance with Lets Encrypt
+
+This section configures [Lets Encrypt](https://letsencrypt.org/) to issue the certificate that Applicaiton Gateway will use for SSL/TLS termination at the gateway.  This uses the Kubernetes add-on [cert-manager](https://github.com/jetstack/cert-manager) to automate the creation and management of the kubernetes secret containing the certificate.
+
+> [!NOTE]
```
Comment (akshaysngupta) [inline docs/tutorial.md]: same

**Comment by akshaysngupta** [comment]
@khowling Amazing! Thanks a lot for your contribution!
Can you please apply these minor fixes?

File: docs/tutorial.md (hunk 2 above)
Comment (khowling) [reply docs/tutorial.md]: I'll add some text about the namespaces Issuers can reside in, and add a reference to the Issuer cert-manager help page that documents all the supported properties

File: docs/tutorial.md (hunk 3 above)
Comment (khowling) [reply docs/tutorial.md]: thanks :)

**Pull request merged**